### PR TITLE
Improve semantics of form checkboxes for assistive technology

### PR DESCRIPTION
**Pull request by pattiereaves**
Fixes #

#### Changes proposed in this Pull Request:

* Puts the checkboxes in a list for assistive technology
* Adds a header above the list for assistive technology

#### Testing instructions:

* The checkboxes in the subscription form should be in a list
* That list should have a header which labels the list.
* When viewing the subscription options underneath the comments box on a site, the check boxes should not see bullets.
image

#### Proposed changelog entry for your changes:

Improves the semantic markup of the checkboxes to subscribe to a comment,

### CODE REVIEW

File: modules/subscriptions.php [hunk 1]
```diff
@@ -630,11 +630,14 @@ function comment_subscribe_init() {
 
 		// Check if Mark Jaquith's Subscribe to Comments plugin is active - if so, suppress Jetpack checkbox
 
-		$str = '';
+		$unique_id = esc_attr( uniqid( 'jetpack-subscription-form-' ) );
```
Comment (zinigor) [inline modules/subscriptions.php]: There's no need to call `esc_attr`, you're doing it properly already by escaping late - when using the value for output.

File: modules/subscriptions.php [hunk 2]
```diff
@@ -630,11 +630,14 @@ function comment_subscribe_init() {
...
+
+		// Add heading and wrap in list for assistive technology
+		$str = '<div><h4 id="'. esc_attr( $unique_id ).'" class="comment-subscription-form-title">Comment subscription preferences</h4><ul class="comment-subscription-form-list" aria-labelledby="'. esc_attr( $unique_id ).'">';
```
Comment (zinigor) [inline modules/subscriptions.php]: The "Comment subscription preferences" header needs to be translated, please use the gettext function for this:
```
__( 'Comment subscription preferences', 'jetpack' )
```

**Review by zinigor — CHANGES_REQUESTED** [review]
This is great overall, thank you! I have added two comments about minor changes that are still needed here.

**Comment by pattiereaves** [comment]
@zinigor I made the changes you requested; ready for a new review.

File: modules/subscriptions.php [hunk 3]
```diff
@@ -630,11 +630,15 @@ function comment_subscribe_init() {
 
 		// Check if Mark Jaquith's Subscribe to Comments plugin is active - if so, suppress Jetpack checkbox
 
-		$str = '';
+		// Create unique ID for the header label
+		$unique_id = uniqid( 'jetpack-subscription-form-' );
+
+		// Add heading and wrap in list for assistive technology
+		$str = '<div><h4 id="'. esc_attr( $unique_id ).'" class="comment-subscription-form-title">'.__( 'Comment subscription preferences', 'jetpack' ).'</h4><ul class="comment-subscription-form-list" aria-labelledby="'. esc_attr( $unique_id ).'">';
```
Comment (zinigor) [inline modules/subscriptions.php]: Please make sure the concatenation operator is separated from the strings with a space on both ends. Maybe this long line can be split into several lines too for greater readibility:
```
$str =
    '<div><h4 id="'
    . esc_attr( $unique_id )
    . '" class="comment-subscription-form-title">'
    . __( 'Comment subscription preferences', 'jetpack' )
    // and so on
```

Sorry I didn't notice it right away. The other fixes look good to me, thank you!

**Pull request closed without merging**